sdk/src/types/block/output/rent.rs

```diff
@@ -140,3 +148,50 @@ impl<T: Rent, const N: usize> Rent for [T; N] {
         self.iter().map(|elem| elem.weighted_bytes(config)).sum()
     }
 }
+
+pub struct MinimumStorageDeposit {
+    config: RentStructure,
+    token_supply: u64,
+    builder: BasicOutputBuilder,
+}
+
+impl MinimumStorageDeposit {
+    pub fn new(config: RentStructure, token_supply: u64) -> Self {
+        Self {
+            config,
+            token_supply,
+            builder: BasicOutputBuilder::new_with_amount(Output::AMOUNT_MIN).add_unlock_condition(
+                AddressUnlockCondition::new(Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH]))),
+            ),
+        }
+    }
+
+    pub fn with_native_tokens(mut self, native_tokens: impl Into<Option<NativeTokens>>) -> Self {
+        if let Some(native_tokens) = native_tokens.into() {
+            self.builder = self.builder.with_native_tokens(native_tokens);
+        }
+        self
+    }
+
+    pub fn with_storage_deposit_return(mut self, return_address: Address) -> Result<Self, Error> {
+        self.builder = self
+            .builder
+            .add_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                return_address,
+                Output::AMOUNT_MIN,
+                self.token_supply,
+            )?);
+        Ok(self)
+    }
+
+    pub fn with_expiration(mut self, return_address: Address) -> Result<Self, Error> {
```

Comment:
Why accept an address here and not also do `Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH]))`?

Reply:
Yeah you are right, blind copy paste, will change